File: ios/Classes/SkyWayRemoteVideoChild.swift
//
//  SkyWayRemoteVideoChild.swift
//  flutter_skyway
//
//  Created by 笠井貴史 on 2020/05/29.
//

import SkyWay
import Foundation
import Flutter

public class SkyWayRemoteVideoChild: NSObject {
    init(remoteStream:SKWMediaStream) {
        self.remoteStream = remoteStream
        remoteStreamView = SKWVideo()
    }
    
    public var remoteStream:SKWMediaStream?
    public var remoteStreamView:SKWVideo
    
    func setUpRemoteStream(){
        remoteStream!.addVideoRenderer(remoteStreamView, track: 0)
    }
    
   public func close() {//描画を止める
        if (nil == remoteStream) {
            return;
        }
        tearDownRemoteStream();
    }

    private func tearDownRemoteStream() {
        remoteStream!.removeVideoRenderer(remoteStreamView, track: 0);
        remoteStream!.close();
    }
}

File: android/src/main/java/com/kasai/flutter_skyway/SkyWayRemoteVideoChild.java
package com.kasai.flutter_skyway;


import io.flutter.plugin.common.PluginRegistry;
import io.skyway.Peer.Browser.Canvas;
import io.skyway.Peer.Browser.MediaStream;

public class SkyWayRemoteVideoChild {

    SkyWayRemoteVideoChild(MediaStream remoteStream,PluginRegistry.Registrar registrar){
        this.remoteStream = remoteStream;
        this.remoteStreamView = new Canvas(registrar.context());
    }

    public MediaStream remoteStream;
    public Canvas remoteStreamView;

    void setUpRemoteStream() {//描画を始める
        remoteStream.addVideoRenderer(remoteStreamView, 0);
    }

    void close() {
        if (null == remoteStream) {
            return;
        }
        tearDownRemoteStream();
    }

    private void tearDownRemoteStream() {
        remoteStream.removeVideoRenderer(remoteStreamView, 0);
        remoteStream.close();
    }
}